src/pages/index.js

```diff
@@ -156,13 +162,45 @@ const editAvatarObject = {
         editProfileAvatarForm.close();
         editProfileAvatarForm.popupForm.reset();
         editAvatarFormValidator.toggleButtonState();
+      })
+      .finally(() => {
         editProfileAvatarForm.renderLoading(false);
       })
       .catch((res) => {
         console.log(`There is an error in the program: ${res}`);
       });
   },
 };
+
+const editAvatarObjectHandleSubmit = {
+  popupSelector: "#edit-profile-avatar-modal",
+  handleFormSubmit: (inputValues) => {
+    handleSubmit(
+      () => {
+        api
```

Comment:
It doesn't work because you nee to **return** the promise here with `return`, `renderLoading` and `catch`


This is the correct code. You don't need to duplicate `close`, 

```js
const editAvatarObjectHandleSubmit = {
  popupSelector: "#edit-profile-avatar-modal",
  handleFormSubmit: (inputValues) => {
    handleSubmit(
      () => {
        return api.updateProfilePicture(inputValues.avatar).then(() => {
          userProfile.setUserAvatar(inputValues.avatar);
          editProfileAvatarForm.popupForm.reset();
          editAvatarFormValidator.toggleButtonState();
        });
      },
      editProfileAvatarForm,
      "Saving..."
    );
  },
};

```